examples/clipboard.rs

```diff
@@ -56,37 +65,64 @@ fn clipboard_copy(mut text: ResMut<CustomText>) {
     });
 }
 
-fn clipboard_paste(mut text: ResMut<CustomText>) {
-    let _task = spawn_local(async move {
-        let window = web_sys::window().expect("window"); // { obj: val };
+#[derive(Default)]
+struct ClipboardChannel {
+    pub rx: Option<Receiver<String>>,
+}
 
-        let nav = window.navigator();
-        let clipboard = nav.clipboard();
+fn setup_clipboard_paste(
+    mut commands: Commands,
+    windows: Query<Entity, With<Window>>,
+    winit_windows: NonSendMut<WinitWindows>,
+    mut clipboardChannel: NonSendMut<ClipboardChannel>,
+) {
+    let Some(first_window) = windows.iter().next() else {
+        return;
+    };
+    let Some(winit_window_instance) = winit_windows.get_window(first_window) else {
+        return;
+    };
+    let (tx, rx): (Sender<String>, Receiver<String>) = mpsc::channel();
 
-        let Ok(permissions) = nav.permissions() else {
-            return;
-        };
-        let clipboard_permission_desc = js_sys::Object::new();
-        js_sys::Reflect::set(
-            &clipboard_permission_desc,
-            &"name".into(),
-            // 'clipboard-read' fails on firefox because it's not implemented.
-            // more info: https://developer.mozilla.org/en-US/docs/Web/API/Permissions_API#browser_compatibility
-            &"clipboard-read".into(),
-        );
-        dbg!(permissions.query(&clipboard_permission_desc.into()));
-        match clipboard {
-            Some(a) => {
-                let p = a.read();
-                let result = wasm_bindgen_futures::JsFuture::from(p)
-                    .await
-                    .expect("clipboard populated");
-                dbg!("result: ", &result);
-                info!("clippyboy worked");
+    use wasm_bindgen::closure::Closure;
+    use wasm_bindgen::prelude::*;
+    use winit::platform::web::WindowExtWebSys;
+    let canvas = winit_window_instance.canvas();
+    info!("canvas found");
+    let closure = Closure::<dyn FnMut(_)>::new(move |event: web_sys::ClipboardEvent| {
+        tx.send("test".to_string());
+        match event
+            .clipboard_data()
+            .expect("could not get clipboard data.")
+            .get_data("text/plain")
+        {
+            Ok(data) => {
+                tx.send(data);
             }
-            None => {
-                warn!("failed to read clippyboy");
+            _ => {
+                info!("Not implemented.");
             }
-        };
+        }
+        info!("{:?}", event.clipboard_data())
     });
+
+    canvas
+        .add_event_listener_with_callback("paste", closure.as_ref().unchecked_ref())
+        .expect("Could not edd paste event listener.");
```

Comment:
I believe this callback doesn't work, I'm not 100% sure. ; I'm not sure how to debug in more details the exact stack trace when paste-ing.

Reply:
https://github.com/mvlabat/bevy_egui/pull/178#discussion_r1195238342